xonsh/cli_utils.py

```diff
@@ -51,58 +94,289 @@ def get_doc(func: tp.Callable, parameter: str = None):
     str
         doc of the parameter/function
     """
+    if isinstance(func, str):
+        return func
     import inspect
 
     doc = inspect.getdoc(func) or ""
     if parameter:
-        return _get_param_doc(doc, parameter)
+        return inspect.cleandoc(_get_param_doc(doc, parameter)).strip()
     else:
-        return _get_func_doc(doc)
+        return _get_func_doc(doc).strip()
 
 
 _FUNC_NAME = "_func_"
 
 
+def _get_args_kwargs(annot: tp.Any) -> tp.Tuple[tp.Sequence[str], tp.Dict[str, tp.Any]]:
+    args, kwargs = [], {}
+    if isinstance(annot, tuple):
+        args, kwargs = annot
+    elif TYPING_ANNOTATED_AVAILABLE and "Annotated[" in str(annot):
+        if hasattr(annot, "__metadata__"):
+            args, kwargs = annot.__metadata__[0]
+        else:
+            from typing_extensions import get_args
```

Comment:
This will fail if `typing_extensions` is not installed and `typing.Annotated` is avaliable

Reply:
I will check it.

Reply:
actually the __metadata__ attribute is available in `typing.Annotated` of py3.9 
the class Annotated is loosely based on that thing. 
for polyfilling previous version one already needs typing_extensions.

Reply:
I see.
So for clarity (and like we do in other files in xonsh), use `get_args` from typing if its available or from `typing_extensions` when it isn't.
Also try importing instead of checking hasattr(tp, "Annotated"). That way we can be sure both `Annotated` and `get_args` exist.

```python
try:
    from typing import Annotated, get_args  # noqa

    TYPING_ANNOTATED_AVAILABLE = True
except ImportError:
    try:
        from typing_extensions import Annotated, get_args  # type: ignore

        TYPING_ANNOTATED_AVAILABLE = True
    except ImportError:
         ...
```

Reply:
get_args is used only when typing_extensions is used. we don't have to import that from `typing`.

>  try importing instead of checking hasattr(tp, "Annotated").

I don't have any opinion on this. I am updating.

Reply:
Also, get_args is py3.8+ I think.

Reply:
You didn't update this 😕

Reply:
Actually I have removed the `if` statement hasattr check and used `importerror`. Here it relies on the fact that Annotated from stdlib has __metadata__ on it.

Reply:
@daniel-shimon I've tested without typing-extension on Python3.9 . It doesn't raise any errors. Let me know if there is any other issues.